documentation/Working_on_Bot/README.md

```diff
@@ -0,0 +1,257 @@
+```shell
+
+## Stage 1: stop current bot
+# enter the command `.exit` on your discord guild
+
+## Stage 2: update base image
+# if you made any new changes to the either of the files specified below,
+ - wall_e/src/requirements.txt file
+ - CI/server_scripts/Dockerfile.base file
+# run the following commands:
+export CONTAINER_HOME_DIR=/usr/src/app;
+export COMPOSE_PROJECT_NAME="project_name"
+ ./CI/user_scripts/create-dev-docker-image.sh
+
+
+## Stage 3
+./CI/user_scripts/deploy-to-test-server.sh;
+```
+
+#### view logs in active time
+```shell
+ docker logs -f "${COMPOSE_PROJECT_NAME}_wall_e"
+```
+
+### Without the Database
+
+#### Step 1. Re-creating the docker base image
+You will need to recreate the base docker image if you
+ * made changes to the wall_e/src/requirements.txt file or
+ * made changes to the CI/server_scripts/Dockerfile.base file
+
+Commands To Run
+```shell
+export CONTAINER_HOME_DIR=/usr/src/app;
+export COMPOSE_PROJECT_NAME="project_name"
+./CI/user_scripts/create-dev-docker-image.sh
+```
+
+#### Step 2. Launching the Bot
+
+```shell
+export ENVIRONMENT=LOCALHOST;
+//ensure that DB_ENBLED is set to 0 via whatever method you want
+export COMPOSE_PROJECT_NAME="project_name"
+
+# if you did not need to re-create the base image you can use
+export ORIGIN_IMAGE="sfucsssorg/wall_e"
+# otherwise, you will need to use the below commmand
+export ORIGIN_IMAGE="${COMPOSE_PROJECT_NAME}_wall_e_base"
+
+./CI/user_scripts/deploy-to-test-server-nodb.sh;
+```
+
+
+#### Re-launching the bot after making changes
+```shell
+
+## Stage 1: stop current bot
+# enter the command `.exit` on your discord guild
+
+## Stage 2: update base image
+# if you made any new changes to the either of the files specified below,
+ - wall_e/src/requirements.txt file
+ - CI/server_scripts/Dockerfile.base file
+# run the following commands:
+export CONTAINER_HOME_DIR=/usr/src/app;
+export COMPOSE_PROJECT_NAME="project_name"
+ ./CI/user_scripts/create-dev-docker-image.sh
+
+
+ ## Stage 3
+./CI/user_scripts/deploy-to-test-server-nodb.sh;
+
+```
+
+#### view logs in active time
+```shell
+ docker logs -f "${COMPOSE_PROJECT_NAME}_wall_e"
+```
+
+## Testing the bot
+
+### Step 1. Run through the [linter](https://en.wikipedia.org/wiki/Lint_%28software%29)
+
+Before you can push your changes to the wall_e repo, you will first need to make sure it passes the unit tests. that can be done like so:
+
+```shell
+./CI/user_scripts/test_walle.sh
+```
+
+### Step 2. Testing on [CSSS Bot Test Server](https://discord.gg/85bWteC)
+After you have tested on your own Discord Test Server, Create a PR to the [Wall-E Repo](https://github.com/CSSS/wall_e/pulls) that follows the [rules](https://github.com/CSSS/wall_e/wiki/4.-Making-a-PR-to-master) for PRs before pushing your changes to [Wall-E](https://github.com/CSSS/wall_e). Creating the PR will automatically load it into the CSSS Bot Test Server. the name of the channel will be `pr-<PR number>`.  
+
+## Test Cases  
```

Comment:
This README is not the best place to list all the test cases. Move it to the wiki or some other README.

Reply:
the reason I placed it here was because my thinking was that the wiki get less frequently updated than the readmes on this repo and the test case would be updated frequently, hence why it is here.

Reply:
fixed with 9aead60ec4abdf235d2d25d41e48f3af39a959d0

Reply:
So you've moved the test cases to a separate README? Could you link to it here?

Reply:
fixed with 0efbb7f9a7fd93cf94c1b51fe428ba92e9e54825